DbgProvider/public/Debugger/DbgEngDebugger.cs

```diff
@@ -1033,6 +1033,26 @@ internal bool QueryTargetIs32Bit()
                 } );
         } // end QueryTargetIs32Bit()
 
+        internal bool QueryTargetIsWow64()
+        {
+            return ExecuteOnDbgEngThread( () =>
+            {
+                CheckHr(m_debugControl.GetActualProcessorType(out IMAGE_FILE_MACHINE type ));
+                if( type == IMAGE_FILE_MACHINE.AMD64 || (uint)type == 0xAA64 /*ARM64*/ ) //Wow64 supported architectures
+                {
+                    CheckHr(m_debugSystemObjects.GetCurrentThreadTeb( out var teb ));
+                    var wow64offset = teb + 0x1488; //TODO: TlsSlots[1] - 1 == WOW64_TLS_CPURESERVED
```

Comment:
>teb + 0x1488 [](start = 38, length = 12)

This does not seem like a good idea. Even if the TEB size / layout never changed, what if the current process is an actual, real 64-bit process: in that case, somebody could be using that TLS slot, and we'll calculate the answer based on what is effectively random data.

The first idea that I had for answering the question "is it a wow64 target" would be to just check if wow64.dll is loaded. Anybody could name their binary wow64.dll, I guess, but you could also check the full path.

I asked around and somebody suggested that you could check if the currently executing code is wow by calling `GetActualProcessorType` and `GetExecutingProcessorType` (or potentially `GetEffectiveProcessorType` depending on what you’re doing). That won't work if the process is currently executing 64-bit code, but depending on why you are interested in wow-ness, maybe that's enough.

Another option might be to check the architecture of the main module, but I'm not sure off the top of my head how to do that.

I heard another suggestion to look at the `WowTebOffset` field, but it looks undocumented, and new to Win10, so I don't know if that'd actually be helpful.

Reply:
I borrowed this approach from wow64exts, so someone on the debugger team seems to have thought it was good enough... but I am inclined to agree it seems less than great.

DbgEng itself seems to have favored a variation on your approach - `ProcessInfo::MayBeWow64x86` checks if kernel32.dll is a folder named "syswow64".

WowTebOffset definitely doesn't do the trick; at least in Win 7 it's `SpareUlong0` and contains 0 for Wow64 processes (though I could check if the TIB ExceptionList is a pointer to the TEB address + 8192, but that doesn't feel like it's getting things any less janky)

The other idea I had was checking whether $ntwsym is populated, with a different value than $ntnsym. Still seems less than idea but at least it would be more directly tied to the assumptions being made about the implications of Wow64.

Checking the architecture of the main module seems like it could be a good idea... I think I'll take a look at that.

Reply:
`GetExecutingProcessorType` definitely doesn't work for (Memory dumps of Wow64 processes from the Task Manager context menu option have an executing processor type of AMD64)